starlite/contrib/sqlalchemy/repository.py

```diff
@@ -511,3 +516,7 @@ def _filter_select_by_kwargs(self, statement: SelectT, **kwargs: Any) -> SelectT
         for key, val in kwargs.items():
             statement = statement.where(getattr(self.model_type, key) == val)
         return statement
+
+    def _base_select(self, **kwargs: Any) -> Select[tuple[ModelT]]:
+        statement: Select[tuple[ModelT]] | None = kwargs.pop("base_select", None)
```

Comment:
You can express this using walrus -

```python

if statement := kwargs.pop("base_select", None):

```

Reply:
I think it can actually be even easier.  I changed it to this:
```python
    def _base_select(self, **kwargs: Any) -> Select[tuple[ModelT]]:
        return kwargs.pop("base_select", self.statement)
```